Src/IronPython/Runtime/PythonRange.cs

```diff
@@ -5,115 +5,132 @@
 // Copyright (c) Pawel Jasinski.
 //
 
+#nullable enable
+
 using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
 using System.Numerics;
 
 using IronPython.Runtime.Binding;
 using IronPython.Runtime.Operations;
 using IronPython.Runtime.Types;
 
 using Microsoft.Scripting.Runtime;
+using Microsoft.Scripting.Utils;
+
+using NotNullAttribute = Microsoft.Scripting.Runtime.NotNullAttribute;
 
 namespace IronPython.Runtime {
-    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1710:IdentifiersShouldHaveCorrectSuffix")]
     [PythonType("range")]
     [DontMapIEnumerableToContains]
-    public sealed class PythonRange : IEnumerable<int>, ICodeFormattable, IList, IReversible {
-        private int _length;
-
-        public PythonRange(object stop) : this(0, stop, 1) { }
-        public PythonRange(object start, object stop) : this(start, stop, 1) { }
-
-        public PythonRange(object start, object stop, object step) {
-            Initialize(start, stop, step);
-        }
-
-        private void Initialize(object ostart, object ostop, object ostep) {
-            // TODO: https://github.com/IronLanguages/ironpython3/issues/472 - should not throw but it's better than an incorrect result...
-            stop = Converter.ConvertToIndex(ostop, throwOverflowError: true);
-            start = Converter.ConvertToIndex(ostart, throwOverflowError: true);
-            step = Converter.ConvertToIndex(ostep, throwOverflowError: true);
-            if (step == 0) {
+    public sealed class PythonRange : ICodeFormattable, IReversible, IEnumerable<int>, IEnumerable<BigInteger> {
```

Comment:
I ended up going with `IEnumerable<int>` just because it's far more common to iterate in that range and for back compat (though I guess in Python 2 range produces a list so thinking about it again that argument is moot). It should have no impact in the Python world since it should all be going through `__iter__`. From the .NET point of view it's quite uncommon to work with `BigInteger`. Anyway, we can always change it later if it's an issue.

Reply:
> From the .NET point of view it's quite uncommon to work with `BigInteger`

Agree. My only concern from the .NET point of view is that after #52 I will be preaching the "Python's `int` type is `BigInteger`" gospel and this will be an exception. But with clear release/migrate notes this should not be an issue.